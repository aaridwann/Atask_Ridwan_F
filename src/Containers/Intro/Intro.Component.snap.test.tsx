import * as React from "react";
import { render, waitFor } from "@testing-library/react-native";

import IntroComponent from "./Intro.Component";

jest.mock("expo-local-authentication");

// jest.mock("expo-local-authentication", () => ({
//   supportedAuthenticationTypesAsync: jest.fn(),
//   authenticateAsync: jest.fn(),
// }));

describe("Snap test", () => {
  const mockValue = {
    available: [0, 1, 2],
    selected: 2,
  };

  const Props = {
    isAuth: false,
    navigation: {
      push: jest.fn(),
    },
  };

  it("Should render with correct props", async () => {
    jest
      .spyOn(React, "useState")
      .mockImplementation(() => [mockValue, jest.fn()]);

    await waitFor(() => {
      expect(render(<IntroComponent {...Props} />)).toMatchSnapshot();
    });
  });
});
