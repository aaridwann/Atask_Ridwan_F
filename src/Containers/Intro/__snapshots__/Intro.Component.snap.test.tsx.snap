// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snap test Should render with correct props 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#283618",
      "display": "flex",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
          "padding": 4,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "marginHorizontal": "auto",
          }
        }
      />
      <Text
        style={
          {
            "color": "white",
            "letterSpacing": 2,
            "marginVertical": 35,
          }
        }
      >
        Login first
      </Text>
      <Text
        style={
          {
            "color": "white",
            "letterSpacing": 2,
            "marginVertical": 35,
          }
        }
      >
        Welcome to
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
